src/main/java/jenkins/plugins/http_request/HttpRequestStep.java

```diff
@@ -426,7 +433,11 @@ FilePath resolveUploadFile() {
 		}
 
 		public Item getProject() {
-			return run.getParent();
+			try {
+				return Objects.requireNonNull(getContext().get(Run.class)).getParent();
+			} catch (IOException | InterruptedException e) {
+				throw new IllegalStateException(e);
+			}
```

Comment:
The general pattern in other Jenkins plugins that I looked at is to allow these exceptions to propagate through `SynchronousNonBlockingStepExecution#run()` (which is declared in the interface with `throws Exception`) without wrapping them. I'm not sure if anything relies on these being wrapped or unwrapped, so to be safe why don't we stick with the established pattern? I think you can do this by just declaring that this method and `HttpRequestExecution#from` throw `IOException` and `InterruptedException`.

Reply:
The method didn't declare any exception previously, so I wasn't sure whether something could. I'm perfectly fine with explicit declarations! :+1:

Reply:
PR updated.